iothub_client/src/iothub_client_core.c

```diff
@@ -837,21 +837,15 @@ static int ScheduleWork_Thread(void* threadArgument)
 static IOTHUB_CLIENT_RESULT StartWorkerThreadIfNeeded(IOTHUB_CLIENT_CORE_INSTANCE* iotHubClientInstance)
 {
     IOTHUB_CLIENT_RESULT result;
-    if (iotHubClientInstance->TransportHandle == NULL)
```

Comment:
>if (iotHubClientInstance->TransportHandle == NULL) [](start = 4, length = 50)

Is there a bug with more details motivating this change?  I haven't been around the TransportHandle code that much but it has special rules.  

You should make sure Ewerton is involved in review since he knows this best.

Reply:
Yes, this change fix the issue #491

---
In reply to: [328724419](https://github.com/Azure/azure-iot-sdk-c/pull/1211#discussion_r328724419) [](ancestors = 328724419)

Reply:
When we a developer creates the iothub from transport, this function will not create the ScheduleWork_Thread, and the convenience layer will never call the IoTHubClientCore_LL_DoWork.

---
In reply to: [328747981](https://github.com/Azure/azure-iot-sdk-c/pull/1211#discussion_r328747981) [](ancestors = 328747981,328724419)

Reply:
But I agree with you, and I will wait for @ewertons

---
In reply to: [328749333](https://github.com/Azure/azure-iot-sdk-c/pull/1211#discussion_r328749333) [](ancestors = 328749333,328747981,328724419)